typescript/testsuite/testsuite_impl/basic_datastore_and_api_test/basic_datastore_and_api_test_.ts

```diff
@@ -0,0 +1,238 @@
+		log.info("Incremented number of books read");
+
+		log.info("Retrieving test person to verify number of books read...");
+		const getPersonResult: Result<Person, Error> = await apiClient.getPerson(TEST_PERSON_ID);
+		if (!getPersonResult.isOk()) {
+			return err(getPersonResult.error);
+		}
+		const person: Person = getPersonResult.value;
+		log.info("Retrieved test person");
+
+		if (person.getBookRead() !== TEST_NUM_BOOKS_READ) {
+			return err(new Error("Expected number of book read '"+TEST_NUM_BOOKS_READ+"' !== actual number of books read '"+person.getBookRead()+"'"));
+		}
+
+		return ok(null);
+	}
+}
+
+// ====================================================================================================
+//                                       Private helper functions
+// ====================================================================================================
+
+//TODO TODO TODO (Ali) - review these helper methods after making final changes to network_impl
+async function getDatastoreServiceConfigurations(): Promise<[ContainerCreationConfig, (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => Result<ContainerRunConfig, Error>]> {
+	const datastoreContainerCreationConfig: ContainerCreationConfig = await getDataStoreContainerCreationConfig();
+
+	const datastoreRunConfigFunc: (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => Result<ContainerRunConfig, Error> = await getDataStoreRunConfigFunc();
+	return [datastoreContainerCreationConfig, datastoreRunConfigFunc];
+}
+
+async function getDataStoreContainerCreationConfig(): Promise<ContainerCreationConfig> {
+	const containerCreationConfig: ContainerCreationConfig = new ContainerCreationConfigBuilder( //TODO (Ali) - may need to make ContainerCreationConfig async
+		DATASTORE_IMAGE,
+	).withUsedPorts(
+		new Set(""+DATASTORE_PORT+"/tcp"),
+	).build()
+	return containerCreationConfig;
+}
+
+async function getDataStoreRunConfigFunc(): Promise<(ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => Result<ContainerRunConfig, Error>> {
+	const runConfigFunc: (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => Result<ContainerRunConfig, Error> = 
+	(ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => {
+		return ok(new ContainerRunConfigBuilder().build());
+	}
+	return runConfigFunc;
+}
+
+async function getApiServiceConfigurations(datastoreClient: DatastoreClient): Promise<[ContainerCreationConfig, (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => Result<ContainerRunConfig, Error>]> {
+	const configInitializingFunc: (fp: number) => Promise<Result<null, Error>> = await getApiServiceConfigInitializingFunc(datastoreClient);
```

Comment:
Same deal - none of this should need to be `async` since it's all fine to be synchronous